src/services/infra/InfraService.ts

```diff
@@ -229,6 +260,159 @@ export default class InfraService {
     return ok(dnsRecord)
   }
 
+  launchSiteFromCms({
+    siteName,
+    primaryDomain,
+    useWww,
+    email,
+  }: LaunchSiteFromCMSParams): ResultAsync<
+    SiteLaunchCreateParams,
+    AmplifyError | SiteLaunchError | MissingUserEmailError
+  > {
+    // prepare site launch params
+    return ResultAsync.fromPromise(
+      this.usersService.findByEmail(email),
+      (error) =>
+        new MissingUserEmailError(
+          `User with email ${email} not found: ${error}`
+        )
+    ).andThen((requestor) => {
+      if (!requestor) {
+        return errAsync(
+          new MissingUserEmailError(`User with email ${email} not found`)
+        )
+      }
+
+      // since site launch in CMS is triggered by the users themselves, we
+      // treat requestor and agency as the same user
+      const user = requestor
+      const agency = requestor
+      const subDomainSettings = [
+        {
+          branchName: "master",
+          prefix: `${useWww ? "www" : ""}`,
+        },
+      ]
+
+      /**
+       * We update the database first to indicate that the user
+       * has started the site launch flow. This prevents users
+       * from retrying and abusing the system.
+       *
+       * The ability to retry after a failed launch is
+       * tracked in IS-273
+       */
+      return this.launchesService.updateDbForLaunchStart(siteName).andThen(() =>
+        ResultAsync.fromPromise(
+          this.launchSite(
+            user,
+            agency,
+            siteName,
+            primaryDomain,
+            subDomainSettings
+          ),
+          (error) => new SiteLaunchError(`Error launching site: ${error}`)
+        ).andThen((siteLaunch) => {
+          if (siteLaunch.isOk()) {
+            return okAsync(siteLaunch.value)
+          }
+          return errAsync(siteLaunch.error)
+        })
+      )
+    })
+  }
+
+  getGeneratedDnsRecords = (
+    siteName: string
+  ): ResultAsync<
+    DnsResultsForSite,
+    MissingSiteError | AmplifyError | SiteLaunchError
+  > =>
+    this.sitesService
+      .getBySiteName(siteName)
+      .andThen(() => this.launchesService.getDNSRecords(siteName))
+
+  getSiteLaunchStatus(
```

Comment:
feels like there should be a function called `getSiteStatusFromDnsRecord` that does `DnsResultForSite` -> `SiteLaunchDto` because that seems to be a repeated pattern

Reply:
on a side note, `DnsResultForSite` sounds weird - wdyt about `SiteDnsResults` ._.